shunting-yard-replicator/src/main/java/com/hotels/shunting/yard/replicator/exec/receiver/ContextFactory.java

```diff
@@ -107,13 +115,26 @@ public Context createContext(MetaStoreEvent event) {
 
   private CircusTrainConfig generateConfiguration(MetaStoreEvent event) {
     String sourceMetaStoreUri = getSourceMetaStoreUri(event);
-    String replicaTableLocation = workOutReplicaLocation(event.getDatabaseName(), event.getTableName());
+
+    String replicaDatabaseName = event.getDatabaseName();
+    String replicaTableName = event.getTableName();
+
+    TableReplication tableReplication = shuntingYardTableReplications
+        .getTableReplicationsMap()
```

Comment:
make the `getTableReplicationsMap` private and add a method `shuntingYardTableReplications.getTableReplication(String databaseName, String tableName)`

encapsulate all the things! + the joining the db and table name will be done in one class

Reply:
done